### PR TITLE
resource_manager: only init the default resource group once

### LINKED ISSUE #6787: resource_manager: default resource group's config is reset after restart pd server
### What did you do?
Deploy a nightly pd server add execute:

```
Alter resource group default ru_per_sec = 1000000 background = (task_name = "br");
```

And then restart the pd server

### What did you expect to see?
```
mysql> select * from information_schema.resource_groups;
+---------+------------+----------+-----------+-------------+-----------------+
| NAME    | RU_PER_SEC | PRIORITY | BURSTABLE | QUERY_LIMIT | BACKGROUND      |
+---------+------------+----------+-----------+-------------+-----------------+
| default | 1000000    | MEDIUM   | YES       | NULL        | TASK_NAMES='br' |
+---------+------------+----------+-----------+-------------+-----------------+
```

### What did you see instead?
```
mysql> select * from information_schema.resource_groups;
+---------+------------+----------+-----------+-------------+------------+
| NAME    | RU_PER_SEC | PRIORITY | BURSTABLE | QUERY_LIMIT | BACKGROUND |
+---------+------------+----------+-----------+-------------+------------+
| default | UNLIMITED  | MEDIUM   | YES       | NULL        | NULL       |
+---------+------------+----------+-----------+-------------+------------+
```

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve?

Issue Number: Close #6787

### What is changed and how does it work?

```
Only init the default resource group once if it is not exist to avoid reset the existing default group's config.
```

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Code changes

- Has the configuration change
- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
None.
```

### CODE REVIEW

File: tests/integrations/mcs/resourcemanager/resource_manager_test.go [hunk 1]
```diff
@@ -940,6 +940,21 @@ func (suite *resourceManagerClientTestSuite) TestBasicResourceGroupCURD() {
 			re.Equal(1, len(groups1))
 		}
 	}
+
+	// test restart cluster
```
Comment (HuSharp) [inline tests/integrations/mcs/resourcemanager/resource_manager_test.go:944]: I don't think this test reflects the pr fix, maybe it's best to change the default rg to ensure consistency.

**Review by nolouch — APPROVED** [review]
lgtm

**Pull request merged**